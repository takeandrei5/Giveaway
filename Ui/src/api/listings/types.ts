import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
