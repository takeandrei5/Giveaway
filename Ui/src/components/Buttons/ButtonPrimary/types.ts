import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
