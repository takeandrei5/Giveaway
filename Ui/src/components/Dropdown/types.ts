export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
