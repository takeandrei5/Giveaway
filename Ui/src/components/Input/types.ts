export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
