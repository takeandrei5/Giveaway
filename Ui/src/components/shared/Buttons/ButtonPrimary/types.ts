import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
