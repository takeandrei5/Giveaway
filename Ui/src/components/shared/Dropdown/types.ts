export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
