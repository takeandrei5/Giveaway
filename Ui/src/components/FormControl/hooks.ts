import { FieldMetaProps } from 'formik';
import { useEffect, useState } from 'react';

import { useCheckFormIsInvalidProps } from './types';

const useCheckFormIsInvalid = (meta: FieldMetaProps<unknown>): useCheckFormIsInvalidProps => {
	const [isInvalid, setIsInvalid] = useState(false);

	useEffect(() => {
		setIsInvalid(meta.touched && !!meta.error);
	}, [meta.touched, meta.error]);

	return { isInvalid };
};

export { useCheckFormIsInvalid };
