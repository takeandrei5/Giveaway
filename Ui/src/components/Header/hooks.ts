import { NextRouter, useRouter } from 'next/router';
import { useCallback } from 'react';

const useLogin = () => {
	const router: NextRouter = useRouter();

	const handleSignIn: () => Promise<boolean> = useCallback(
		async () => await router.replace('/api/auth/login'),
		[router]
	);
	const handleSignInWithReturnTo: (returnTo: string) => Promise<boolean> = useCallback(
		async (returnTo: string) => await router.replace(`/api/auth/login?returnTo=${returnTo}`),
		[router]
	);
	const handleSignOut: () => Promise<boolean> = useCallback(() => router.replace('/api/auth/logout'), [router]);

	return { handleSignIn, handleSignInWithReturnTo, handleSignOut };
};

export { useLogin };
