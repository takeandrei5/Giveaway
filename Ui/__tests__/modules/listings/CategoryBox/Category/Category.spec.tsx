import { Category } from '@modules/listings/CategoryBox/Category';
import { CategoryProps } from '@modules/listings/CategoryBox/Category/types';
import { useAppDispatch, useAppSelector } from '@redux/hooks';
import { CategoryState } from '@redux/slices/changeCategorySlice';
import { EmptyObject } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

jest.mock('@redux/hooks', () => ({
	useAppDispatch: jest.fn(),
	useAppSelector: jest.fn((categoryState) => ({})),
}));

describe('Category', () => {
	it('should match snapshot', () => {
		// Arrange
		const props: CategoryProps = {
			category: 'books',
			name: 'test',
			image: 'https://via.placeholder.com/150',
		};

		// Act
		const { container } = render(<Category {...props} />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	it('should trigger useCallback everytime the category box is clicked', async () => {
		// Arrange
		const props: CategoryProps = {
			category: 'books',
			name: 'test',
			image: 'https://via.placeholder.com/150',
		};

		(useAppSelector as unknown as jest.Mock).mockImplementation((state) => ({ category: 'books' }));

		(useAppDispatch as unknown as jest.Mock).mockImplementation(() => jest.fn());

		// Act
		const { getByTestId } = render(<Category {...props} />);
		await userEvent.click(getByTestId('category-box'));

		// Assert
		expect(useAppDispatch).toBeCalledTimes(1);
	});
});
