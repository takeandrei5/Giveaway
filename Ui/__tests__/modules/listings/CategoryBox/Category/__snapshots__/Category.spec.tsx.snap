// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should match snapshot 1`] = `
<div>
  <div
    class="css-8atqhb"
  >
    <div
      class="css-1gihwm6"
      data-testid="category-box"
    >
      <div
        class="chakra-stack css-1jxjhsf"
      >
        <div
          class="css-1oa8n4a"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              alt="test"
              data-nimg="responsive"
              decoding="async"
              draggable="false"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <span
          class="chakra-text css-muf1uf"
          data-testid="typography"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;
